app/src/main/java/org/simple/clinic/teleconsultlog/prescription/patientinfo/TeleconsultPatientInfoView.kt

```diff
@@ -0,0 +1,122 @@
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.teleconsultlog.prescription.TeleconsultPrescriptionScreenKey
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.util.unsafeLazy
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import javax.inject.Inject
+import javax.inject.Named
+
+class TeleconsultPatientInfoView constructor(
+    context: Context,
+    attrs: AttributeSet?
+) : CardView(context, attrs), TeleconsultPatientInfoUi {
+
+  @Inject
+  lateinit var userClock: UserClock
+
+  @Inject
+  @Named("full_date")
+  lateinit var dateFormatter: DateTimeFormatter
+
+  @Inject
+  lateinit var effectHandler: TeleconsultPatientInfoEffectHandler
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  private val delegate by unsafeLazy {
+    val screenKey = screenRouter.key<TeleconsultPrescriptionScreenKey>(this)
+    val uiRenderer = TeleconsultPatientInfoUiRenderer(this)
+
+    MobiusDelegate.forView(
+        events = Observable.never(),
+        defaultModel = TeleconsultPatientInfoModel.create(patientUuid = screenKey.patientUuid),
+        init = TeleconsultPatientInfoInit(),
+        update = TeleconsultPatientInfoUpdate(),
+        effectHandler = effectHandler.build(),
+        modelUpdateListener = uiRenderer::render
+    )
+  }
+
+  init {
+    inflate(context, R.layout.view_teleconsult_patient_info, this)
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    delegate.start()
+  }
+
+  override fun onDetachedFromWindow() {
+    delegate.stop()
+    super.onDetachedFromWindow()
+  }
+
+  override fun onSaveInstanceState(): Parcelable? {
+    return delegate.onSaveInstanceState(super.onSaveInstanceState())
+  }
+
+  override fun onRestoreInstanceState(state: Parcelable?) {
+    super.onRestoreInstanceState(delegate.onRestoreInstanceState(state))
+  }
+
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) return
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun renderPatientInformation(patientProfile: PatientProfile) {
+    displayPatientNameGenderAge(patientProfile.patient)
+    displayPatientAddress(patientProfile.address)
+  }
+
+  private fun displayPatientNameGenderAge(patient: Patient) {
+    val ageValue = DateOfBirth.fromPatient(patient, userClock).estimateAge(userClock)
+    val patientGender = patient.gender
+
+    patientNameTextView.text = context.getString(
+        R.string.view_teleconsult_patient_info,
+        patient.fullName,
+        context.getString(patientGender.displayLetterRes),
+        ageValue.toString()
+    )
+  }
+
+  private fun displayPatientAddress(address: PatientAddress) {
+    val addressFields = listOf(
+        address.streetAddress,
+        address.colonyOrVillage,
+        address.district,
+        address.state,
+        address.zone
+    ).filterNot { it.isNullOrBlank() }
+
+    patientAddressTextView.text = addressFields.joinToString()
```

Comment:
Is this bit of code that joins the address fields into a comma-separated string repeated in a few places? I feel like I've seen similar pieces of code earlier.

Reply:
If yes, we might just want to extract it to a method on the `PatientAddress` class.

Reply:
Yes, will do that.